mage_ai/orchestration/pipeline_scheduler.py

```diff
@@ -317,9 +331,13 @@ def queued_block_runs(self) -> List[BlockRun]:
                 completed = block is not None and \
                     block.all_upstream_blocks_completed(completed_block_uuids)
 
+            failed_block_uuids = set(b.block_uuid for b in self.failed_block_runs)
             if completed:
                 block_run.update(status=BlockRun.BlockRunStatus.QUEUED)
                 queued_block_runs.append(block_run)
+            elif any(b in failed_block_uuids for b in block.upstream_block_uuids):
+                block_run.update(
```

Comment:
Why do we update status in this method?

Reply:
because we were already iterating through the block runs and updating status, but yeah it's probably not the best place for this